airbyte-server/src/main/java/io/airbyte/server/ServerApp.java

```diff
@@ -210,15 +210,49 @@ public static void main(String[] args) throws Exception {
       LOGGER.info(String.format("Setting Database version to %s...", airbyteVersion));
       jobPersistence.setVersion(airbyteVersion);
     }
-    final Optional<String> airbyteDatabaseVersion = jobPersistence.getVersion();
+
+    Optional<String> airbyteDatabaseVersion = jobPersistence.getVersion();
+    if (airbyteDatabaseVersion.isPresent() && !AirbyteVersion
+        .isCompatible(airbyteVersion, airbyteDatabaseVersion.get())
+        && !isDatabaseVersionAheadOfAppVersion(airbyteVersion, airbyteDatabaseVersion.get())) {
+      runAutomaticMigration(configRoot, configRepository, jobPersistence, airbyteVersion,
+          airbyteDatabaseVersion.get());
+      // After migration, upgrade the DB version
+      airbyteDatabaseVersion = jobPersistence.getVersion();
+    }
 
     if (airbyteDatabaseVersion.isPresent() && AirbyteVersion.isCompatible(airbyteVersion, airbyteDatabaseVersion.get())) {
       LOGGER.info("Starting server...");
       new ServerApp(configRepository, jobPersistence, configs).start();
     } else {
-      LOGGER.info("Start serving version mismatch errors...");
+      LOGGER.info("Start serving version mismatch errors. Automatic migration must have failed");
```

Comment:
> We have one last check to see if the process was successful. If yes then we allow the server to start, if no then we set the database version back to initial version if we had changed it and then start the server with VersionMismatch error.

in the case of error while automatically migrating is running, should the old config volume be restored, so airbyte could still be useable if it was rollbacked with the older version?